File: ts-sdk/src/sdk/models/shared/retentionpolicyassignments.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { RetentionPolicyAssignmentBase } from "./retentionpolicyassignmentbase";



// RetentionPolicyAssignments
/** 
 * The part of an API response that describes marker
 * based pagination
**/
export class RetentionPolicyAssignments extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=entries", elemType: RetentionPolicyAssignmentBase })
  entries?: RetentionPolicyAssignmentBase[];

  @SpeakeasyMetadata({ data: "json, name=limit" })
  limit?: number;

  @SpeakeasyMetadata({ data: "json, name=next_marker" })
  nextMarker?: string;
}

File: java-sdk/lib/src/main/java/models/shared/RetentionPolicyAssignments.java
package .models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * RetentionPolicyAssignments
 * The part of an API response that describes marker
 * based pagination
**/
public class RetentionPolicyAssignments {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("entries")
    public RetentionPolicyAssignmentBase[] entries;
    public RetentionPolicyAssignments withEntries(RetentionPolicyAssignmentBase[] entries) {
        this.entries = entries;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("limit")
    public Long limit;
    public RetentionPolicyAssignments withLimit(Long limit) {
        this.limit = limit;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("next_marker")
    public String nextMarker;
    public RetentionPolicyAssignments withNextMarker(String nextMarker) {
        this.nextMarker = nextMarker;
        return this;
    }
}